src/plugins/condition/components/inspector/StylesView.vue

```diff
@@ -139,54 +159,135 @@ export default {
             conditionsLoaded: false,
             navigateToPath: '',
             selectedConditionId: '',
-            locked: false
+            items: [],
+            domainObject: undefined
         };
     },
     computed: {
+        locked() {
+            // TODO should this be for parent layouts only?
+            return this.selection.some(selectionPath => {
+                const self = selectionPath[0].context.item;
+                const parent = selectionPath.length > 1 ? selectionPath[1].context.item : undefined;
+
+                return (self && self.locked) || (parent && parent.locked);
+            });
+        },
         allowEditing() {
             return this.isEditing && !this.locked;
+        },
+        isMainLayoutSelected() {
+            const selectedItem = this.selection[0][0].context.item;
+            const selectedLayoutItem = this.selection[0][0].context.layoutItem;
+
+            return selectedItem && selectedItem.type === 'layout' && !selectedLayoutItem;
+        },
+        styleableFontItems() {
+            return this.selection.filter(selectionPath => {
+                const item = selectionPath[0].context.item;
+                const itemType = item && item.type;
+                const layoutItem = selectionPath[0].context.layoutItem;
+                const layoutItemType = layoutItem && layoutItem.type;
+                const NON_STYLEABLE_CONTAINER_TYPES = [
+                    'layout',
+                    'flexible-layout',
+                    'tabs'
+                ];
+                const NON_STYLEABLE_LAYOUT_ITEM_TYPES = [
+                    'line-view',
+                    'box-view',
+                    'image-view'
+                ];
```

Comment:
Since these values dont change, can we set them in data, so we are not creating new variables everytime this is called

Reply:
moved these out of the function block